openedx/openedx-platform
# ENT-162 Create an EnterpriseCourseEnrollment during the enrollment flow

This PR introduces creation of EnterpriseCourseEnrollments during the LMS enrollment workflow. To do so, I've introduced an EnterpriseApiClient to the `util` app that's used to perform requests against the Enterprise API.

This change is heavily dependent on https://github.com/edx/edx-enterprise/pull/40, and it can be best tested by triggering the enrollment view from ecommerce using these instructions: https://github.com/edx/ecommerce/pull/1127

**JIRA tickets**: ENT-162

**Discussions**:

* Originally, this logic was [implemented in Ecommerce](https://github.com/edx/ecommerce/pull/1122), but it was later decided to move this into the enrollment view.
* After a final conversation with @mattdrayer related to this comment, we agreed to pipe consent through the enrollment view: https://github.com/edx/edx-enterprise/pull/40#issuecomment-280391520

**Dependencies**: https://github.com/edx/edx-enterprise/pull/40, ENT-134 

**Sandbox URL**: TBD - sandbox is being provisioned.

**Testing instructions**:

1. Setup a devstack.
2. Check out this edx-platform branch.
3. Follow the testing instructions in this PR: https://github.com/edx/ecommerce/pull/1127

**Reviewers**

**Settings**
```yaml
EDXAPP_FEATURES:
  ENABLE_COMBINED_LOGIN_REGISTRATION: true
```
